esp/src/src-react/components/WorkunitDetails.tsx

```diff
@@ -54,20 +54,24 @@ export const WorkunitDetails: React.FunctionComponent<WorkunitDetailsProps> = ({
 
     const [logCount, setLogCount] = React.useState<number | string>("*");
     const [logsDisabled, setLogsDisabled] = React.useState(true);
+
+    const fetchLogCount = React.useCallback(() => {
+        service.GetLogsEx({ ...queryParams, jobId: wuid, LogLineStartFrom: 0, LogLineLimit: 10 }).then(response => {    // HPCC-27711 - Requesting LogLineLimit=1 causes issues
+            setLogCount(response.total);
+        });
+    }, [queryParams, wuid]);
+
     React.useEffect(() => {
         hasLogAccess().then(response => {
             setLogsDisabled(!response);
             return response;
         }).then(hasLogAccess => {
             if (hasLogAccess) {
-                service.GetLogsEx({ ...queryParams, jobId: wuid, LogLineStartFrom: 0, LogLineLimit: 10 }).then(response => {    // HPCC-27711 - Requesting LogLineLimit=1 causes issues
-                    setLogCount(response.total);
-                });
+                fetchLogCount();
```

Comment:
Rather than calling `fetchLogAccess` here, you could make it a `useEffect` instead of a `useCallback`?

Reply:
Switching to the more detailed comparison provided in useDeepEffect only calls GetLogsEx when queryParams genuinely changes, rather than 3-5 times as was the case with useEffect.